File: c++/프로그래머스 3진법 뒤집기.cpp
#include <string>
#include <vector>
#include <cmath>

using namespace std;

int solution(int n) {
    int answer = 0;
    vector<int> three;
    while(1)
    {
        if(n==0)
            break;
        three.push_back(n%3);
        n = n/3;
    }
    for(int i=three.size()-1; i>=0; i--)
    {
        answer= answer + three[i]*pow(3, three.size()-1-i);
    }
    return answer;
}

File: python/프로그래머스 3진법 뒤집기.py
import queue

def solution(n):
    answer = 0
    arr = queue.LifoQueue()
    while n != 0:
        arr.put(n%3)
        n = n//3
    count = 0
    while not arr.empty():
        answer = answer + arr.get()*(3**count)
        count = count+1
    return answer